internal/kafka/test/integration/kafka_expiration_test.go

```diff
@@ -0,0 +1,513 @@
+								mocksupportedinstancetypes.WithLifespanSeconds(&longLifespanSeconds),
+							),
+						},
+						SupportedBillingModels: []config.KafkaBillingModel{
+							{
+								ID:               types.STANDARD.String(),
+								AMSResource:      ocm.RHOSAKResourceName,
+								AMSProduct:       string(ocm.RHOSAKProduct),
+								AMSBillingModels: []string{string(amsv1.BillingModelStandard)},
+							},
+						},
+					},
+					{
+						Id:          types.DEVELOPER.String(),
+						DisplayName: types.DEVELOPER.String(),
+						Sizes: []config.KafkaInstanceSize{
+							*mocksupportedinstancetypes.BuildKafkaInstanceSize(
+								mocksupportedinstancetypes.WithLifespanSeconds(&shortLifespanSeconds),
+							),
+						},
+						SupportedBillingModels: []config.KafkaBillingModel{
+							{
+								ID:               types.STANDARD.String(),
+								AMSResource:      ocm.RHOSAKResourceName,
+								AMSProduct:       string(ocm.RHOSAKTrialProduct),
+								AMSBillingModels: []string{string(amsv1.BillingModelStandard)},
+							},
+						},
+					},
+				},
+			},
+		}
+	})
+	defer tearDown()
+
+	mockKasFleetshardSyncBuilder := kasfleetshardsync.NewMockKasFleetshardSyncBuilder(h, t)
+	mockKasfFleetshardSync := mockKasFleetshardSyncBuilder.Build()
+	mockKasfFleetshardSync.Start()
+	defer mockKasfFleetshardSync.Stop()
+
+	// set up dummy data plane cluster that accepts both standard and developer clusters
+	testCluster := mockclusters.BuildCluster(func(cluster *api.Cluster) {
+		cluster.Meta = api.Meta{
+			ID: api.NewID(),
+		}
+		cluster.ProviderType = api.ClusterProviderStandalone // ensures no errors will occur due to it not being available on ocm
+		cluster.SupportedInstanceType = api.AllInstanceTypeSupport.String()
+		cluster.ClientID = "some-client-id"
+		cluster.ClientSecret = "some-client-secret"
+		cluster.ClusterID = "test-cluster"
+		cluster.CloudProvider = mocks.MockCloudProviderID
+		cluster.Region = mocks.MockCloudRegionID
+		cluster.Status = api.ClusterReady
+		cluster.ProviderSpec = api.JSON{}
+		cluster.ClusterSpec = api.JSON{}
+		cluster.AvailableStrimziVersions = api.JSON(mockclusters.AvailableStrimziVersions)
+	})
+	db := h.DBFactory().New()
+	err := db.Create(testCluster).Error
+	g.Expect(err).NotTo(gomega.HaveOccurred(), "failed to create dummy data plane cluster")
+
+	// ensure kafka request with long lifespan does not get deleted
+	account := h.NewRandAccount()
+	ctx := h.NewAuthenticatedContext(account, nil)
+	result, resp, err := client.DefaultApi.CreateKafka(ctx, true, public.KafkaRequestPayload{
+		Name: "kafka-with-long-lifespan",
+		Plan: fmt.Sprintf("%s.x1", types.STANDARD), // has a long lifespan of 2d
+	})
+	if resp != nil {
+		_ = resp.Body.Close()
+	}
+	g.Expect(err).ToNot(gomega.HaveOccurred(), "failed to create kafka with long lifespan")
+	expectedExpirationDate := result.CreatedAt.Add(time.Duration(longLifespanSeconds) * time.Second).Format(time.RFC1123Z)
```

Comment:
Why this timestamp format?

Reply:
Can't the times be directly compared?

Reply:
for some reason, the time format outputted in the tests is different from my local one. So I formatted it to ensure that both dates are the same. The tests were failing the pr checks otherwise.

Reply:
I think this is fixable. But it can be investigated in a future change